Src/Test.UnitTests.Sarif.PatternMatcher/ValidationResultTests.cs

```diff
@@ -0,0 +1,434 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+using FluentAssertions;
+
+using Microsoft.RE2.Managed;
+
+using Moq;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher
+{
+    public class ValidationResultTests
+    {
+        [Fact]
+        public void SearchSkimmer_ValidationResultsOverrideIndexAndLength()
+        {
+            // test using OverrideIndexTestValidator
+            string ruleName = "OverrideIndexTest";
+            string validatorAssemblyPath = $@"c:\{Guid.NewGuid()}.dll";
+            string scanTargetExtension = Guid.NewGuid().ToString();
+
+            var mockFileSystem = new Moq.Mock<IFileSystem>();
+            mockFileSystem.Setup(x => x.FileExists(validatorAssemblyPath)).Returns(true);
+            mockFileSystem.Setup(x => x.AssemblyLoadFrom(validatorAssemblyPath)).Returns(this.GetType().Assembly);
+
+            var validators = new ValidatorsCache(
+                new string[] { validatorAssemblyPath },
+                fileSystem: mockFileSystem.Object);
+
+            string regexStr = "(?im)^.*?(?P<refine>\\bTestTerm\\b).*?$";
+            string fileContent = "TestTerm Another-TEST-TERM";
+            MatchExpression expression =
+                CreateMatchExpression(
+                    regexStr,
+                    allowFileExtension: scanTargetExtension);
+
+            SearchDefinition definition = CreateSearchDefinition(ruleName, expression);
+
+            AnalyzeContext context =
+                CreateSkimmer(
+                    scanTargetExtension: scanTargetExtension,
+                    definition,
+                    out SearchSkimmer skimmer,
+                    fileContent,
+                    validators: validators);
+
+            skimmer.Analyze(context);
+
+            IList<Result> results = ((TestLogger)context.Logger).Results;
+            results.Should().NotBeNull();
+            results.Count.Should().Be(1);
+            results[0].Kind.Should().Be(ResultKind.Fail);
+            results[0].Level.Should().Be(FailureLevel.Warning);
+            results[0].Locations.Should().NotBeNull();
+            results[0].Locations.Count.Should().Be(1);
+            results[0].Locations[0].PhysicalLocation.Should().NotBeNull();
+            results[0].Locations[0].PhysicalLocation.Region.Should().NotBeNull();
+
+            // region points to 2nd uppercase "testterm"
+            Region region = results[0].Locations[0].PhysicalLocation.Region;
+            region.CharOffset.Should().Be(17);
```

Comment:
>17 [](http://example.com/codeflow?start=42&length=2)

how those were defined?
can u create like a text where we can see how this was generated?
for example, if thats  TestTerm, so something like TestTerm.length

this is to simplify future changes :)

Reply:
thanks